public/app/features/explore/ContentOutline/ContentOutline.tsx

```diff
@@ -50,70 +41,218 @@ export function ContentOutline({ scroller, panelId }: { scroller: HTMLElement |
       behavior: 'smooth',
     });
 
     reportInteraction('explore_toolbar_contentoutline_clicked', {
       item: 'select_section',
-      type: buttonTitle,
+      type: itemPanelId,
     });
   };
 
   const toggle = () => {
-    toggleExpanded();
+    toggleContentOutlineExpanded();
     reportInteraction('explore_toolbar_contentoutline_clicked', {
       item: 'outline',
-      type: expanded ? 'minimize' : 'expand',
+      type: contentOutlineExpanded ? 'minimize' : 'expand',
+    });
+  };
+
+  const toggleSection = () => {
+    toggleSectionExpanded();
+    reportInteraction('explore_toolbar_contentoutline_clicked', {
+      item: 'section',
+      type: sectionExpanded ? 'minimize' : 'expand',
     });
   };
 
   useEffect(() => {
-    const activeItem = outlineItems.find((item) => {
-      const top = item?.ref?.getBoundingClientRect().top;
+    let activeItem;
 
-      if (!top) {
-        return false;
+    for (const item of outlineItems) {
+      let top = item?.ref?.getBoundingClientRect().top;
+
+      // Check item
+      if (top && top >= 0) {
+        activeItem = item;
       }
 
-      return top >= 0;
-    });
+      // Check children
+      const activeChild = item.children?.find((child) => {
+        const offsetTop = child.customTopOffset || 0;
+        let childTop = child?.ref?.getBoundingClientRect().top;
+        return childTop && childTop >= offsetTop;
+      });
 
-    if (!activeItem) {
-      return;
-    }
+      if (activeChild) {
+        setActiveSectionChildId(activeChild.id);
+        setActiveSectionId(item.id);
+        break;
+      }
 
-    setActiveItemId(activeItem.id);
-  }, [outlineItems, verticalScroll]);
+      if (activeItem) {
+        setActiveSectionId(activeItem.id);
+        setActiveSectionChildId(undefined);
+        break;
+      }
+    }
+  }, [outlineItems, verticalScroll, sectionExpanded]);
 
   return (
     <PanelContainer className={styles.wrapper} id={panelId}>
       <CustomScrollbar>
         <div className={styles.content}>
           <ContentOutlineItemButton
-            title={expanded ? 'Collapse outline' : undefined}
-            icon={expanded ? 'angle-left' : 'angle-right'}
+            icon={'arrow-from-right'}
+            tooltip={contentOutlineExpanded ? 'Collapse outline' : 'Expand outline'}
+            tooltipPlacement={contentOutlineExpanded ? 'right' : 'bottom'}
             onClick={toggle}
-            tooltip={!expanded ? 'Expand content outline' : undefined}
-            className={styles.buttonStyles}
-            aria-expanded={expanded}
+            className={cx(styles.toggleContentOutlineButton, {
+              [styles.justifyCenter]: !contentOutlineExpanded && !outlineItemsShouldIndent,
+            })}
+            aria-expanded={contentOutlineExpanded}
+            width={contentOutlineExpanded ? 160 : 40}
           />
 
-          {outlineItems.map((item) => {
-            return (
+          {outlineItems.map((item) => (
```

Comment:
Not super happy with the complexity of content outline, but I had to address multiple states like collapsed, expanded, indented child and indented parent. Furthermore, making the truncation work (ellipsis), has required some custom button widths.

Reply:
Cleaned up custom button widths thanks to @Elfo404 😄